File: Services/mqtt-service.py
from os import path
import json
from datetime import datetime
from time import sleep
import paho.mqtt.client as mqtt
from influx import Influx

influx = Influx()

f = open(path.abspath(path.join(path.dirname(__file__), '../MQTT/mqtt-service-config.json')), 'r')
config = json.load(f)
f.close()

def on_connect(client, userdata, flags, rc):
  print('[OK  ] connected with result code {:d}'.format(rc))
  client.subscribe(config['topic'])

def on_message(client, userdata, msg):
  data_dict = json.loads(msg.payload.decode())
  print('[LOG ]\n{:s}'.format(json.dumps(data_dict, indent=2)))
  influx.write_sensor(data_dict)

client = mqtt.Client(client_id="mqtt_service_eva")
client.username_pw_set(config['user'], config['password'])
client.on_connect = on_connect
client.on_message = on_message

connected = False
while not connected:
  try:
    print('[WAIT] connecting')
    client.connect(config['server'], 1883, 60)
    connected = True
  except:
    pass
  sleep(1)

client.loop_forever()

File: Services/mqtt-service.js
const fs = require('fs');
const path = require('path');
const mqtt = require('mqtt');
const influx = require('./influx');

const config = JSON.parse(fs.readFileSync(path.join(__dirname, '../MQTT/mqtt-service-config.json'), 'utf-8'));

const client  = mqtt.connect(config.url, {
  username: config.user,
  password: config.password,
  clientId: config.id,
  reconnectPeriod: 2000
});

client.on('connect', (response) => {
  console.log('[OK  ] connection')
  console.log('[WAIT] subscribe');
  client.subscribe(config.topic, (err) => {
    if(err) {
      console.log('[ERR ] subscribe');
    } else {
      console.log('[OK  ] subscribe');
    }
  });
});

client.on('message', (topic, message) => {
  let dataStr = message.toString();
  console.log(`[LOG ] ${dataStr}`);
  influx.writeSensor(dataStr);
});

client.on('error', (error) => {
  console.log(`[ERR ] connection -> ${error.message}`);
});

console.log('[WAIT] connection');
